gradio/data_classes.py

```diff
@@ -11,9 +11,31 @@
 
 from fastapi import Request
 from gradio_client.utils import traverse
-from pydantic import BaseModel, RootModel, ValidationError
 from typing_extensions import Literal
 
+from . import wasm_utils
+
+if not wasm_utils.IS_WASM:
+    from pydantic import BaseModel, RootModel, ValidationError  # type: ignore
+else:
+    # XXX: Currently Pyodide V2 is not available on Pyodide,
+    # so we install V1 for the Wasm version.
+    from pydantic import BaseModel as BaseModelV1
+    from pydantic import ValidationError
+
+    # Map V2 method calls to V1 implementations.
+    # Ref: https://docs.pydantic.dev/latest/migration/#changes-to-pydanticbasemodel
+    class BaseModel(BaseModelV1):
+        pass
+
+    BaseModel.model_dump = BaseModel.dict  # type: ignore
+    BaseModel.model_json_schema = BaseModel.schema  # type: ignore
+
+    # RootModel is not available in V1, so we create a dummy class.
+    class RootModel(BaseModel):
+        def __init__(__pydantic_self__, root=None, **data) -> None:  # type: ignore
+            pass
+
```

Comment:
Replacing all the V2 usages in the codebase with V1 API like `from pydantic.v1 import BaseModel` as described in https://docs.pydantic.dev/latest/migration/#continue-using-pydantic-v1-features is also an option.

However, I have chosen an option of monkey-patching the V2 classes like this, because I thought Pydantic V2 would probably be supported on Pyodide in the future.

WDYT?

Reply:
Nice